maven-shared-doc-resource/src/main/resources/markdown/download.md.vm

```diff
@@ -0,0 +1,54 @@
+---
+title: Download ${project.name} Source
+---
+#*
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*#
+#set( $h = "#" )
+#set( $downloadDirectory = $project.properties.downloadDirectory )
+#if( ! $downloadDirectory )
+#set( $downloadDirectory = "plugins" )
+#end
+#set( $downloadArtifactId = $project.properties.downloadArtifactId )
+#if( ! $downloadArtifactId )
+#set( $downloadArtifactId = $project.artifactId )
+#end
+
+# Download ${project.name} ${project.version} Source
+
+${project.name} ${project.version} is distributed in source format\. Use a source archive if you intend to build
+${project.name} yourself\. Otherwise, simply use the ready\-made binary artifacts from central repository\.
+
+In order to guard against corrupted downloads/installations, it is highly recommended to
+[verify the signature](https://www.apache.org/dev/release-signing#verifying-signature)
+of the release bundles against the public [KEYS](https://www.apache.org/dist/maven/KEYS) used by the Apache Maven
+developers\.
+
+${project.name} is distributed under the [Apache License, version 2\.0](https://www.apache.org/licenses/)\.
+
+${h}${h} ${project.name} ${project.version}
+
+This is the current stable version of ${project.name}\.
+
+|  | Link                                                                                                                                                   |Checksum|Signature|
+|---|--------------------------------------------------------------------------------------------------------------------------------------------------------|---|---|
+${project.name} ${project.version} \(Source zip\)| [${downloadArtifactId}\-${project.version}\-source\-release\.zip](\[preferred\]/maven/${downloadDirectory}/${downloadArtifactId}-${project.version}-source-release.zip) |[${downloadArtifactId}\-${project.version}\-source\-release\.zip\.sha512](https://www.apache.org/dist/maven/${downloadDirectory}/${downloadArtifactId}-${project.version}-source-release.zip.sha512)|[${downloadArtifactId}\-${project.version}\-source\-release\.zip\.asc](https://www.apache.org/dist/maven/${downloadDirectory}/${downloadArtifactId}-${project.version}-source-release.zip.asc)|
```

Comment:
Let us use the new download location for checksum and signature: downloads.a.o instead of www.a.o/dist.